File: server/Models/VaultKeeps.cs
namespace keepr.Models;

public class VaultKeeps : RepoItems<int>
{
  public string CreatorId { get; set; }
  public int VaultId { get; set; }
  public int KeepId { get; set; }
}


public class VaultKeepsProfile : Profile
{
  public int VaultKeepsId { get; set; }
  public int VaultId { get; set; }

}


public class VKModel : RepoItems<int>
{
  public int VaultKeepId { get; set; }
  public int VaultId { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Views { get; set; }
  public int Kept { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }

}

File: client/src/models/VaultKeeps.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";

export class VaultKeep extends DatabaseItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.views = data.views
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
    this.kept = data.kept
    this.vaultKeepId = data.vaultKeepId
  }
}